packages/core-react/src/components/Type/index.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+
+export default function Type({ Tag, style, className, ...rest }) {
+  const classes = clsx(className, `ray-text--${style}`);
+
+  return <Tag className={classes} {...rest} />;
+}
+
+Type.propTypes = {
+  Tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
+  style: PropTypes.oneOf([
+    'h1',
```

Comment:
Just curious, how were these types determined?

Reply:
these are all the type options defined in core